relay-server/src/utils/shutdown.rs

```diff
@@ -70,24 +70,34 @@ pub trait FutureExt: Sized {
 
 impl<F> FutureExt for F where F: Sized {}
 
-#[test]
-#[should_panic(expected = "Dropped unfinished future during shutdown: bye")]
-fn test_basic() {
-    use std::time::{Duration, Instant};
-    use tokio_timer::Delay;
+#[cfg(test)]
+mod tests {
+    use actix::clock::Clock;
 
-    System::run(|| {
-        actix::spawn(
-            Delay::new(Instant::now() + Duration::from_millis(100)).then(|_| {
-                System::current().stop();
+    #[test]
+    #[should_panic(expected = "Dropped unfinished future during shutdown: bye")]
+    fn test_basic() {
+        use std::time::Duration;
+
+        use super::*;
+        let mut test_now = relay_test::TestNow::new();
+        let test_clock = Clock::new_with_now(test_now.clone());
+        let sys = System::builder().clock(test_clock);
+
+        sys.run(move || {
+            actix::spawn(relay_test::delay(Duration::from_millis(1000)).then(|_| {
+                println!("Stopping...");
+                actix::System::current().stop();
                 Ok(())
-            }),
-        );
+            }));
 
-        actix::spawn(
-            Delay::new(Instant::now() + Duration::from_millis(200))
-                .drop_guard("bye")
-                .then(|_| Ok(())),
-        );
-    });
+            actix::spawn(
+                relay_test::delay(Duration::from_millis(2000))
+                    .drop_guard("bye")
+                    .then(|_| Ok(())),
+            );
+
+            test_now.advance(Duration::from_millis(1500));
```

Comment:
Now that we're driving the clock manually, can we just create the future, advance a bit, then stop the system without the second delay?

Reply:
Or reduce the test to this, in which case we don't even need the manually driven clock?

```rust
System::run(move || {
    actix::spawn(
        relay_test::delay(Duration::from_secs(10))
            .drop_guard("bye")
            .then(|_| Ok(())),
    );

    actix::System::current().stop();
});
```

Reply:
Yes, that makes sense.